README.md

```diff
@@ -2,34 +2,66 @@
 
 # VMPooler
 
-- [VMPooler](#vmpooler)
-  - [Usage](#usage)
-    - [v2.0.0 note](#v200-note)
-  - [Installation](#installation)
-    - [Dependencies](#dependencies)
-      - [Redis](#redis)
-      - [Other gems](#other-gems)
-  - [Configuration](#configuration)
-  - [Components](#components)
-    - [API](#api)
-    - [Dashboard](#dashboard)
-  - [Related tools and resources](#related-tools-and-resources)
-    - [Command-line Utility](#command-line-utility)
-    - [Vagrant plugin](#vagrant-plugin)
-  - [Development](#development)
-    - [docker-compose](#docker-compose)
-    - [Running docker-compose inside Vagrant](#running-docker-compose-inside-vagrant)
-    - [URLs when using docker-compose](#urls-when-using-docker-compose)
-  - [Update the Gemfile Lock](#update-the-gemfile-lock)
-  - [Releasing](#releasing)
-  - [License](#license)
+- [Usage](#usage)
+  - [Migrating to v3](#migrating-to-v3)
+  - [v2.0.0 note](#v200-note)
+- [Installation](#installation)
+  - [Dependencies](#dependencies)
+    - [Redis](#redis)
+    - [Other gems](#other-gems)
+- [Configuration](#configuration)
+- [Components](#components)
+  - [API](#api)
+  - [Dashboard](#dashboard)
+- [Related tools and resources](#related-tools-and-resources)
+  - [Command-line Utility](#command-line-utility)
+  - [Vagrant plugin](#vagrant-plugin)
+- [Development](#development)
+  - [docker-compose](#docker-compose)
+  - [Running docker-compose inside Vagrant](#running-docker-compose-inside-vagrant)
+  - [URLs when using docker-compose](#urls-when-using-docker-compose)
+- [Update the Gemfile Lock](#update-the-gemfile-lock)
+- [Releasing](#releasing)
+- [License](#license)
 
 VMPooler provides configurable 'pools' of instantly-available (pre-provisioned) and/or on-demand (provisioned on request) virtual machines.
 
 ## Usage
 
 At [Puppet, Inc.](http://puppet.com) we run acceptance tests on thousands of disposable VMs every day. VMPooler manages the life cycle of these VMs from request through deletion, with options available to pool ready instances, and provision on demand.
 
+### Migrating to v3
+
+Starting with the v3.x release, management of DNS records has implemented as DNS plugins, similar to compute providers. This means each pool configuration should be pointing to a configuration object in `:dns_config` to determine it's method of record management.
+
+For those using the global `DOMAIN` environment variable or global `:config.domain` key, this means records were not previously being managed by VMPooler (presumably managed via dynamic dns), so it's value should be moved to `:dns_configs:<INSERT_YOUR_OWN_SYMBOL>:domain` with the value for `dns_class` for the config set to `dynamic-dns`.
+
+For example, the following < v3.x configuration:
+
+```yaml
+:config:
+  domain: 'example.com'
+```
```

Comment:
If this has not been done already, maybe a check could be added in vmpooler.rb that causes startup to fail if this setting is present.

Reply:
Sure that should be an easy add.

Reply:
Done